lang/cs/Org.Apache.REEF.Driver/Defaults/DefaultTaskFailureHandler.cs

```diff
@@ -27,14 +28,30 @@ namespace Org.Apache.REEF.Driver.Defaults
     /// </summary>
     public class DefaultTaskFailureHandler : IObserver<IFailedTask>
     {
+        private static readonly Logger Log = Logger.GetLogger(typeof(DefaultTaskFailureHandler));
+
         [Inject]
         public DefaultTaskFailureHandler()
         {
         }
 
         public void OnNext(IFailedTask value)
         {
-            throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task {0} has failed, and no handler was bound for IFailedTask", value.Id));
+            if (value.Data.IsPresent())
+            {
+                Log.Log(Level.Error, "Task {0} failed with exception", value.Id, value.AsError());
+                throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task {0} has failed on {1}, and no handler was bound for IFailedTask", 
+                    new object[] {value.Id, value.Message}), value.AsError());
+            }
+            else
+            {
+                Log.Log(Level.Error, "Task {0} failed with NO exception information", value.Id);
+                throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task {0} has failed on {1}, and no handler was bound for IFailedTask", 
+                    new object[] {value.Id, value.Message}));
+            }
+
+            throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "Task {0} has failed, and no handler was bound for IFailedTask", value.Id), 
+                value.AsError());
```

Comment:
don't use explicit `new object[]` for `params` arguments. also, make sure your lines are not too long.
oh, and we can now use `$"..."` string interpolation, so:
```C#
if (value.Data.IsPresent())
{
    Log.Log(Level.Error, $"Task {value.Id} failed with exception", value.AsError());
    throw new InvalidOperationException(
        $"Task {value.Id} has failed on {value.Message}, and no handler was bound for IFailedTask", 
        value.AsError());
}
else
{
    Log.Log(Level.Error, "Task {0} failed with NO exception information", value.Id);
    throw new InvalidOperationException(
        $"Task {value.Id} has failed on {value.Message}, and no handler was bound for IFailedTask");
}

throw new InvalidOperationException(
    $"Task {value.Id} has failed, and no handler was bound for IFailedTask", 
    value.AsError());
```
(note that we use `$""` in the first logging occurrence (but *not* the second one) to make sure we log the exception properly)